server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeFilters.java

```diff
@@ -66,9 +65,13 @@ public static DiscoveryNodeFilters buildFromKeyValue(OpType opType, Map<String,
 
     private final OpType opType;
 
+    @Nullable
+    private final DiscoveryNodeFilters trimmed;
```

Comment:
Naming nit: would be good to mention that it's the tier preference that was trimmed - `withoutTierPreferences` or similar perhaps?

Reply:
++ sounds good, renamed